### PR TITLE
Add implementation for 409 conflict

### PR DESCRIPTION
**Pull request by murillo128**

### CODE REVIEW

File: plugins/obs-webrtc/whep-source.cpp [hunk 1]
```diff
@@ -129,14 +99,13 @@ void WHEPSource::Update(obs_data_t *settings)
 	bearer_token =
 		std::string(obs_data_get_string(settings, "bearer_token"));
 
-	if (endpoint_url.empty() || bearer_token.empty()) {
+	if (endpoint_url.empty()) {
```
Comment (murillo128) [inline plugins/obs-webrtc/whep-source.cpp:102]: allow bearer token to be empty

File: plugins/obs-webrtc/whep-source.cpp [hunk 2]
```diff
@@ -16,20 +16,7 @@ const uint8_t naluTypePPS = 8;
 
 WHEPSource::WHEPSource(obs_data_t *settings, obs_source_t *source)
 	: source(source),
-	  endpoint_url(),
```
Comment (murillo128) [inline plugins/obs-webrtc/whep-source.cpp]: No need to empty initialize class attributes

File: plugins/obs-webrtc/whep-source.cpp [hunk 3]
```diff
@@ -16,20 +16,7 @@ const uint8_t naluTypePPS = 8;
...
-	  resource_url(),
-	  bearer_token(),
-	  peer_connection(nullptr),
-	  audio_track(nullptr),
-	  video_track(nullptr),
-	  running(false),
-	  start_stop_mutex(),
-	  start_stop_thread(),
-	  last_frame(std::chrono::system_clock::now()),
-	  last_audio_rtp_timestamp(0),
```
Comment (murillo128) [inline plugins/obs-webrtc/whep-source.cpp]: Initialization to default values is better done on class declaration

File: plugins/obs-webrtc/whep-source.cpp [hunk 4]
```diff
@@ -52,34 +39,17 @@ WHEPSource::WHEPSource(obs_data_t *settings, obs_source_t *source)
 
 WHEPSource::~WHEPSource()
 {
-	running = false;
```
Comment (murillo128) [inline plugins/obs-webrtc/whep-source.cpp]: variable not used anywhere

File: plugins/obs-webrtc/whep-source.cpp [hunk 5]
```diff
@@ -228,7 +197,7 @@ static inline enum speaker_layout convert_speaker_layout(uint8_t channels)
 	}
 }
 
-void WHEPSource::OnFrameAudio(rtc::binary msg, rtc::FrameInfo frame_info)
+void WHEPSource::OnFrameAudio(rtc::binary& msg, const rtc::FrameInfo& frame_info)
```
Comment (murillo128) [inline plugins/obs-webrtc/whep-source.cpp:200]: passing by reference to avoid memory copy of the frame bytes

**Comment by Sean-Der** [comment]
All changes rolled into upstream WHEP PR, thank you @murillo128 !

**Pull request closed without merging**